content/en/blog/_posts/2023-10-24-taint-eviction-controller.md

```diff
@@ -0,0 +1,61 @@
+---
+layout: blog
+title: "Kubernetes 1.29: Decouple TaintManager from NodeLifecycleController"
+date: 2023-11-15
+slug: kubernetes-1-29-taint-eviction-controller
+---
+
+**Authors:** Yuan Chen (Apple), Andrea Tosatto (Apple)
+
+This blog discusses a new feature in Kubernetes 1.29 to improve the handling of taint-based pod eviction.
+
+## Background
+
+In Kubernetes 1.29, an  improvement has been introduced to enhance the handling of taint-based pod eviction on nodes. This blog discusses the changes made to the `NodeLifecycleController` to separate its responsibilities and improve overall code maintainability.
+
+### Summary of Changes
+
+The `NodeLifecycleController` previously combined two independent functions:
+
+- Adding a pre-defined set of `NoExecute` taints to nodes based on the node conditions.
+- Performing pod eviction on NoExecute taints
+
+With the latest release, the taint-based eviction implementation has been moved out of the `node-lifecycle-controller` into a separate and independent component called `TaintEvictionController`. This separation aims to disentangle code, enhance code maintainability, and facilitate future extensions to either component.
```

Comment:
I'd add something high-lighting the benefits and the potential use-cases where this could be useful, as done in the KEP.

Reply:
Added the following 


### Use Cases

This new feature will allow cluster administrators to extend and enhance the default `TaintEvictionController` and even replace the default `TaintEvictionController` with a custom implementation to meet different needs, e.g., better supportof stateful workloads that use `PersistentVolume` on  local disks.

Reply:
Added the metrics description as follows.

As part of the change, addtitonal metrics are introduced to monitor taint-based pod evictionis.

- `pod_deletion_duration_seconds` measures th latency between the time when a taint effect has been activated for the Pod and its deletion via `TaintEvictionController`.
- `pod_deletions_total` reports the total number of Pods deleted by `TaintEvictionController` since its start